cli/crates/backend/src/project.rs

```diff
@@ -1,23 +1,60 @@
-use crate::consts::{DEFAULT_DOT_ENV, DEFAULT_SCHEMA, USER_AGENT};
+use crate::consts::{DEFAULT_DOT_ENV, DEFAULT_SCHEMA_SDL, DEFAULT_SCHEMA_TS, USER_AGENT};
 use crate::errors::BackendError;
 use async_compression::tokio::bufread::GzipDecoder;
 use async_tar::Archive;
-use common::consts::{GRAFBASE_DIRECTORY_NAME, GRAFBASE_ENV_FILE_NAME, GRAFBASE_SCHEMA_FILE_NAME};
-use common::environment::Project;
+use common::consts::{
+    GRAFBASE_DIRECTORY_NAME, GRAFBASE_ENV_FILE_NAME, GRAFBASE_SCHEMA_FILE_NAME, GRAFBASE_SDK_PACKAGE_NAME,
+    GRAFBASE_SDK_PACKAGE_VERSION, GRAFBASE_TS_CONFIG_FILE_NAME,
+};
+use common::environment::{self, Project};
 use http_cache_reqwest::{CACacheManager, Cache, CacheMode, HttpCache};
 use reqwest::{header, Client};
 use reqwest_middleware::ClientBuilder;
 use serde::Deserialize;
-use std::env;
 use std::fs;
 use std::io::{Error as IoError, ErrorKind as IoErrorKind};
 use std::iter::Iterator;
 use std::path::PathBuf;
+use std::{env, fmt};
 use tokio_stream::StreamExt;
 use tokio_util::compat::TokioAsyncReadCompatExt;
 use tokio_util::io::StreamReader;
 use url::Url;
 
+#[derive(Debug, Clone, Copy)]
+pub enum ConfigType {
+    TypeScript,
+    GraphQL,
+}
+
+impl AsRef<str> for ConfigType {
+    fn as_ref(&self) -> &str {
+        match self {
+            ConfigType::TypeScript => "typescript",
+            ConfigType::GraphQL => "graphql",
+        }
+    }
+}
+
+impl fmt::Display for ConfigType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
`f: &mut fmt::Formatter<'_>` https://github.com/rust-lang/rust-clippy/issues/10915

Reply:
This is kind of the convention how Rust does it, how it's documented and how the code action does it: https://doc.rust-lang.org/std/fmt/trait.Display.html

I prefer the `f`, but I also understand if @yoav-lavi doesn't like it.

Reply:
wowie-zowie.